client/core/types.go

```diff
@@ -594,6 +594,7 @@ type PendingFeeState struct {
 // Exchange represents a single DEX with any number of markets.
 type Exchange struct {
 	Host             string                 `json:"host"`
+	Cert             []byte                 `json:"cert"`
```

Comment:
Just noting one more time that this means the frontend has the TLS certificate and could make the XHR request itself.  That said, I'm also fine with the Go html template execution doing `fetchExchangeLogo` as you have it.  It's probably more proper to keep all server communications to the dexc backend.

Reply:
Hmm, one issue with this is if the DEX is down, loading the html page hangs (times out), whereas it could be async on the frontend.

Another issue I'm seeing is that there's no caching of the logo anywhere, so whenever the user loads one of the three pages that would show the logo, it makes a server request.  That's a bit much.

Reply:
> Hmm, one issue with this is if the DEX is down, loading the html page hangs (times out), whereas it could be async on the frontend.

I would set a timeout for the request.

Reply:
> I would set a timeout for the request.

Doesn't matter.  This makes a fast *local* page load have the latency of a remote server.

Also, it's not necessary to involve the config response to achieve caching of the logo.

Another issue not yet mentioned is the network route used to retrieve the logo.  Say the user has a proxy (tor or otherwise) configured for their authenticated server comms, it would be a major information leak to fetch the logo on the open internet.

Reply:
> Another issue not yet mentioned is the network route used to retrieve the logo. Say the user has a proxy (tor or otherwise) configured for their authenticated server comms, it would be a major information leak to fetch the logo on the open internet.

The webserver could have this set when it is started so we can use it when need(which would be just for this).

Reply:
@chappjc, I have re-implemented my suggestion in https://github.com/decred/dcrdex/pull/2004/commits/7838b89117045ce9eaeb7a9503a019e933ac25e3 and I believe it handles major concerns (caching and proxy) adequately. Please take a look and let me know if I missed anything.